estargz/build.go

```diff
@@ -431,19 +430,20 @@ func (tf *tempFiles) TempFile(dir, pattern string) (*os.File, error) {
 	return f, nil
 }
 
-func (tf *tempFiles) CleanupAll() (allErr error) {
+func (tf *tempFiles) CleanupAll() error {
 	tf.filesMu.Lock()
 	defer tf.filesMu.Unlock()
+	var allErr *Errors
 	for _, f := range tf.files {
 		if err := f.Close(); err != nil {
-			allErr = multierror.Append(allErr, err)
+			allErr = allErr.Append(err)
 		}
 		if err := os.Remove(f.Name()); err != nil {
-			allErr = multierror.Append(allErr, err)
+			allErr = allErr.Append(err)
 		}
 	}
 	tf.files = nil
-	return nil
+	return allErr
```

Comment:
We should return nil when no error is appended to allErr?

https://play.golang.org/p/cxUefp-NH9a

Reply:
if nothing is ever appended to it, it is still nil

Reply:
🤔 or you are saying this is the weird Go typed nil nonsense.  blah, let me see...

Reply:
I reworked this ~completely to just use `[]error` directly and have the errorutil package expose an `Aggregate` method for combining them.  I think this ends up being a lot simpler, and it avoid the typed nil case.